openquake/hazardlib/gsim/yu_2013.py

```diff
@@ -0,0 +1,258 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2014-2018 GEM Foundation
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Module exports :class:`YuEtAl2013`, :class:`YuEtAl2013Tibet`,
+:class:`YuEtAl2013Eastern`, :class:`YuEtAl2013Stable`
+
+"""
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+
+from scipy.constants import g
+from scipy.optimize import minimize
+
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import PGA, PGV, SA
+
+
+def gc(coeff, mag):
+    """
+    Returns the set of coefficients to be used for the calculation of GM
+    as a function of earthquake magnitude
+
+    :param coeff:
+        A dictionary of parameters for the selected IMT
+    :param mag:
+        Magnitude value
+    :return:
+        The set of coefficient to be used
+    """
+    if mag > 6.5:
+        a1ca = coeff['ua']
+        a1cb = coeff['ub']
+        a1cc = coeff['uc']
+        a1cd = coeff['ud']
+        a1ce = coeff['ue']
+        a2ca = coeff['ia']
+        a2cb = coeff['ib']
+        a2cc = coeff['ic']
+        a2cd = coeff['id']
+        a2ce = coeff['ie']
+    else:
+        a1ca = coeff['a']
+        a1cb = coeff['b']
+        a1cc = coeff['c']
+        a1cd = coeff['d']
+        a1ce = coeff['e']
+        a2ca = coeff['ma']
+        a2cb = coeff['mb']
+        a2cc = coeff['mc']
+        a2cd = coeff['md']
+        a2ce = coeff['me']
+    return a1ca, a1cb, a1cc, a1cd, a1ce, a2ca, a2cb, a2cc, a2cd, a2ce
+
+
+def rbf(ra, coeff, mag):
+    """
+    Calculate the meadian ground motion for a given magnitude and distance
+
+    :param ra:
+        Distance value [km]
+    :param coeff:
+        A dictionary of parameters for the
+    :param mag:
+        Magnitude value
+    :return:
+
+    """
+    a1ca, a1cb, a1cc, a1cd, a1ce, a2ca, a2cb, a2cc, a2cd, a2ce = gc(coeff, mag)
+    term1 = a1ca + a1cb * mag + a1cc * np.log(ra + a1cd*np.exp(a1ce*mag))
+    term2 = a2ca + a2cb * mag
+    term3 = a2cd*np.exp(a2ce*mag)
+    return np.exp((term1 - term2) / a2cc) - term3
+
+
+def fnc(ra, *args):
+    """
+    Function used in the minimisation problem.
+
+    :param ra:
+        Semi-axis of the ellipses used in the Yu et al.
+    :return:
```

Comment:
Wrong, it must be :returns:

Reply:
Adjusted. I actually checked a few days ago online guidelines for docstrings; it is not clear (to me) if 'return' or 'returns' should be used.